src/drivers/indexeddb.js

```diff
@@ -994,42 +1019,49 @@ function dropInstance(options, callback) {
                     return;
                 }
 
-                var newVersion = db.version + 1;
+                const newVersion = db.version + 1;
 
                 _deferReadiness(options);
 
-                var dbContext = dbContexts[options.name];
-                var forages = dbContext.forages;
+                const dbContext = dbContexts[options.name];
+                const forages = dbContext.forages;
 
-                for (var i = 0; i < forages.length; i++) {
-                    var forage = forages[i];
-                    if (forage._dbInfo.db) {
-                        forage._dbInfo.db.close();
-                        forage._dbInfo.db = null;
-                        forage._dbInfo.version = newVersion;
-                    }
+                db.close();
+                for (let i = 0; i < forages.length; i++) {
+                    const forage = forages[i];
+                    forage._dbInfo.db = null;
+                    forage._dbInfo.version = newVersion;
                 }
 
-                var dropObjectPromise = new Promise(function(resolve, reject) {
-                    var req = idb.open(options.name, newVersion);
+                const dropObjectPromise = new Promise(function(
+                    resolve,
+                    reject
+                ) {
+                    const req = idb.open(options.name, newVersion);
 
-                    req.onerror = reject;
+                    req.onerror = e => {
+                        const db = req.result;
+                        db.close();
+                        reject(e);
+                    };
 
                     req.onupgradeneeded = function() {
                         var db = req.result;
                         db.deleteObjectStore(options.storeName);
                     };
 
                     req.onsuccess = function() {
-                        var db = req.result;
+                        const db = req.result;
+                        db.close();
                         resolve(db);
                     };
                 });
 
                 return dropObjectPromise
                     .then(function(db) {
-                        for (var j = 0; j < forages.length; j++) {
-                            var forage = forages[j];
+                        dbContext.db = db;
+                        for (let j = 0; j < forages.length; j++) {
```

Comment:
Should be `const` too, if we stay with the loop.

Reply:
See earlier comment
> [const  isn't allowed in for loops](https://babeljs.io/repl/#?babili=false&browsers=IE%208&build=&builtIns=usage&code_lz=GYVwdgxgLglg9mABMAFDAlIg3gKEYiBAZzgBsBTAOlLgHM10BuHAXxx0LCKkQC9EAvIgDaAcgCGogDSiARtNERRAXWbA4AJ0QpO3RDEGIADI32IAPH2rkwtKAAtTMANTPMuAJBFyUACowAW3I4ECgUFEwBAD5kBiZWHCA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=true&lineWrap=true&presets=es2015-loose&prettier=false&targets=&version=6.26.0&envVersion=1.6.2)